src/phoenix/server/api/queries.py

```diff
@@ -77,50 +80,88 @@
 from phoenix.server.api.types.UserRole import UserRole
 
 
+@strawberry.input
+class ModelNamesInput:
+    provider_key: GenerativeProviderKey
+
+
 @strawberry.type
 class Query:
     @strawberry.field
-    async def model_providers(
-        self, vendors: List[str], info: Info[Context, None]
-    ) -> List[ModelProvider]:
-        all_vendors = {
-            "OpenAI": ModelProvider(  # https://platform.openai.com/docs/models
-                name="OpenAI",  # currently only models using the chat completions API
-                model_names=[
-                    "o1-preview",
-                    "o1-preview-2024-09-12",
-                    "o1-mini",
-                    "o1-mini-2024-09-12",
-                    "gpt-4o",
-                    "gpt-4o-2024-08-06",
-                    "gpt-4o-2024-05-13",
-                    "chatgpt-4o-latest",
-                    "gpt-4o-mini",
-                    "gpt-4o-mini-2024-07-18",
-                    "gpt-4-turbo",
-                    "gpt-4-turbo-2024-04-09",
-                    "gpt-4-turbo-preview",
-                    "gpt-4-0125-preview",
-                    "gpt-4-1106-preview",
-                    "gpt-4",
-                    "gpt-4-0613",
-                    "gpt-3.5-turbo-0125",
-                    "gpt-3.5-turbo",
-                    "gpt-3.5-turbo-1106",
-                    "gpt-3.5-turbo-instruct",
-                ],
+    async def model_providers(self) -> List[GenerativeProvider]:
+        return [
+            GenerativeProvider(
+                name="OpenAI",
+                key=GenerativeProviderKey.OPENAI,
             ),
-            "Anthropic": ModelProvider(  # https://docs.anthropic.com/en/docs/about-claude/models#model-comparison
-                name="Anthropic",  # currently only models using the messages API
-                model_names=[
-                    "claude-3-5-sonnet-20240620",
-                    "claude-3-opus-20240229",
-                    "claude-3-sonnet-20240229",
-                    "claude-3-haiku-20240307",
-                ],
+            GenerativeProvider(
+                name="Azure OpenAI",
+                key=GenerativeProviderKey.AZURE_OPENAI,
             ),
-        }
-        return [all_vendors[vendor] for vendor in vendors]
+            GenerativeProvider(
+                name="Anthropic",
+                key=GenerativeProviderKey.ANTHROPIC,
+            ),
+        ]
+
+    @strawberry.field
+    async def model_names(self, input: ModelNamesInput) -> List[str]:
+        if (provider_key := input.provider_key) == GenerativeProviderKey.OPENAI:
```

Comment:
not blocking but we might want to just make this a filter eventually

Reply:
Do you mean do the filtering by provider on the frontend? Is the thought to save the extra query when the provider is changed?